File: 367. Valid Perfect Square/367.c
bool isPerfectSquare(int num){
    int start = 0;
    int end = num > 46340 ? 46340 : num;

    while(start<=end){
        int mid = (start+end)/2;
        int tmp = mid*mid;
        if(tmp == num){
            return true;
        }
        if (tmp>num){
            end = mid-1;
        }else{
            start = mid+1;
        }
    }

    return false;
}

File: 367. Valid Perfect Square/367.go
package leetcode

func isPerfectSquare(num int) bool {
	start := 1
	end := num
	for(start <= end){
		mid := (start+end)/2
		tmp := mid*mid
		if tmp == num{
			return true
		}
		if tmp>num{
			end = mid-1
		}else{
			start = mid+1
		}
	}

	return false
}